src/View/TemplateExistsInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link      https://cakephp.org CakePHP(tm) Project
+ * @since     3.0.0
+ * @license   https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Chialab\FrontendKit\View;
+
+/**
+ * Template exists interface.
+ */
+interface TemplateExistsInterface
```

Comment:
Mmmh questa cosa mi suona male, utilizzare un'interfaccia "template exists interface" per dare all'oggetto che la implementa il metodo "templateExists" pare una forzatura.

Mi sembra decisamente più un metodo statico da oggetto factory o qualcosa di simile. È un metodo che - almeno dalla signature - non sembra legato alla definizione dell'oggetto e che può essere slegato dalla sua istanza. Potresti avere tranquillamente `oggettoFactory->templateExists($name)` con il metodo statico.



Poi chiaro, funziona anche così, ma a livello ingegneristico non mi sembra la scelta migliore. Poi magari è una best practice di PHP e fai finta che non ho scritto nulla.

Reply:
Purtroppo non è statico: i template vengono cercati dall'istanza della View, che decide dove cercare i template e che estensione usare. Per forza di cose dobbiamo inserirci nell'istanza della View.

Reply:
Okay, chiaro. 
Allora piccolo appunto personale che però puoi anche ignorare: se non è in programma di riusare quella specifica interfaccia per altre classi che non siano AppView, la casserei e porterei il metodo già implementato in AppView - ovvero, a meno che non pensi di riusarla altrove, non starei ad aggiungere interfacce al calderone "tanto per".

Reply:
La soluzione dell'interfaccia imho è più portabile e più efficiente, perché poi dove hai bisogno di questo metodo puoi semplicemente fare `if ($view instanceof TemplateExistsInterface)` invece di doversi inventare cose più arzigogolate, che comunque non darebbero type-safety come l'interfaccia.
E anche in un progetto in cui per qualche ragione non vuoi usare `AppView` del FK, potresti comunque implementare l'interfaccia ed evitare il `beforeRender` ripetuto.

Reply:
Chiarissimo boys, grazie